File: go/0744_find_smallest_letter_greater_than_target.go
func nextGreatestLetter(letters []byte, target byte) byte {
	if target >= letters[len(letters)-1] {
		return letters[0]
	}

	l, h := 0, len(letters)
	for l < h {
		m := l + (h-l)/2 - 1
		if letters[m] > target {
			h = m
		} else {
			l = m + 1
		}
	}
	return letters[h]
}

File: python/0744_find_smallest_letter_greater_than_target.py
class Solution:
    def nextGreatestLetter(self, letters, target):
        """
        :type letters: List[str]
        :type target: str
        :rtype: str
        """

        return letters[self.bisect_right(letters, target) % len(letters)]

    def bisect_right(self, nums, target):
        lo, hi = 0, len(nums)

        while lo < hi:
            mid = lo + (hi - lo) // 2

            if nums[mid] > target:
                hi = mid
            else:
                lo = mid + 1

        return lo
